### PR TITLE
map more Reviewer information (PUBD-985)

### PR DESCRIPTION
**Pull request by hardyoyo**
- Add the post_process method that @everreau suggested in PUBD-985

### CODE REVIEW

File: serializers.py [hunk 1]
```diff
@@ -384,6 +384,17 @@ def create(self, validated_data: dict) -> Account:
             return existing
         except Account.DoesNotExist:
             return super().create(validated_data)
+        
+    #TODO add a post_process method to create interest objects for the user and
```
Comment (hardyoyo) [inline serializers.py]: Remove TODO comment

**Comment by hardyoyo** [comment]
Currently deployed to dev, ready for testing

**Comment by hardyoyo** [comment]
@justingonder recommends this as a test procedure:

> wipe and re-transport of CPCEM on dev

**Comment by hardyoyo** [comment]
docs on how this is all supposed to work: https://github.com/cdlib/pad-sys-doc/wiki/OJS-to-Janeway-(AKA-Journal-Transporter)

File: serializers.py [hunk 2]
```diff
@@ -381,9 +386,36 @@ def create(self, validated_data: dict) -> Account:
...
+        # to many field.
+        if('interests' in data):
+            interests = data.get("interests").split(",")
+            # filter out empty strings from the interests list, boring
+            interests = list(filter(None, interests))
+            for interest in interests:
+                # first add a new interest object
+                # if it already exists, it will just return the existing one
+                # if it doesn't exist, it will create a new one
+                interest_object = Interest.objects.get_or_create(name=interest)
```
Comment (bibliotechy) [inline serializers.py]: variable unpacking might make this a little more readable.
`interest_object, was_created = Interest.objects.get_or_create(name=interest)`

File: serializers.py [hunk 3]
```diff
@@ -381,9 +386,36 @@ def create(self, validated_data: dict) -> Account:
...
+                # next, save the interest object, just in case
+                interest_object[0].save()
```
Comment (bibliotechy) [inline serializers.py]: I think we only need to save it if it was created. If it wasn't created, then it already exists and isn't being updated, right?

**Comment by hardyoyo** [comment]
image

**Comment by hardyoyo** [comment]
reviewer-interests

**Comment by hardyoyo** [comment]
Two successful journal migration runs for journals with interests that need transporting: uclastat_cts_ties and uciem_cpcem, this code is ready for review, approval and merge

File: serializers.py [hunk 4]
```diff
@@ -260,6 +258,23 @@ def handle_attachments(self, object) -> None:
...
+    def add_interests_to_user(self, user: Account, data: dict) -> None:
+            if('interests' in data):
+                interests = data.get("interests").split(",")
+                # filter out empty strings from the interests list, boring
+                interests = list(filter(None, interests))
+                for interest in interests:
+                    # first add a new interest object
+                    # if it already exists, it will just return the existing one
+                    # if it doesn't exist, it will create a new one
+                    interest_object = Interest.objects.get_or_create(name=interest)
```
Comment (bibliotechy) [inline serializers.py]: I think the idiomatic way to write this is with variable expansion, as in 
```python
interest_object, was_created = Interest.objects.get_or_create(name=interest)
```
That way `interest_object[0]` becomes `interest_object` in all the places you need to use it.

**Review by bibliotechy — APPROVED** [review]
One stylistic comment that you can take or leave.

**Pull request merged**